server/trace/mock_trace_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/empty"
+
+	"google.golang.org/genproto/googleapis/devtools/cloudtrace/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/proto"
+)
+
+const bufSize = 1024 * 1024
+
+var (
+	client cloudtrace.TraceServiceClient
+	conn   *grpc.ClientConn
+	ctx    context.Context
+	lis    *bufconn.Listener
+)
+
+func setup() {
+	// Setup the in-memory server.
+	lis = bufconn.Listen(bufSize)
+	grpcServer := grpc.NewServer()
+	cloudtrace.RegisterTraceServiceServer(grpcServer, &MockTraceServer{})
+	go func() {
+		if err := grpcServer.Serve(lis); err != nil {
+			log.Fatalf("server exited with error: %v", err)
+		}
+	}()
+
+	// Setup the connection and client.
+	ctx = context.Background()
+	var err error
+	conn, err = grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("failed to dial bufnet: %v", err)
+	}
+	client = cloudtrace.NewTraceServiceClient(conn)
+}
+
+func tearDown() {
+	conn.Close()
+}
+
+func TestMain(m *testing.M) {
+	setup()
+	retCode := m.Run()
+	tearDown()
+	os.Exit(retCode)
+}
+
+func bufDialer(context.Context, string) (net.Conn, error) {
+	return lis.Dial()
+}
+
+func generateSpan() *cloudtrace.Span {
+	spanName := fmt.Sprintf("test-span-%d", rand.Intn(100))
```

Comment:
Try to avoid using random for tests, it makes it harder to debug.

Can you change this to `generateSpan(spanName string)`.

Then later on:
```
	spans := []*cloudtrace.Span{
		generateSpan("test-span-1"),
		generateSpan("test-span-2"),
		generateSpan("test-span-3"),
	}
```

So a failure `have "test-span-2", want "test-span-1"` will be easier to figure out.

Reply:
Yep, done